test/unit/asset-fs-inline-path-shadow/output.js

```diff
@@ -96,7 +96,7 @@ console.log(fs.readFileSync(__dirname + '/asset.txt', 'utf8'));
 
 (function () {
   var join = () => 'nope';
-  console.log(fs.readFileSync(join(__dirname, 'asset.txt'), 'utf8'));
+  console.log(fs.readFileSync(join(__dirname + '/asset-fs-inline-path-shadow', 'asset.txt'), 'utf8'));
```

Comment:
This is a consequence of directory emission widening - because there is a `__dirname` reference that can't be analyzed to anything more specific than that, the entire directory is emitted. Note how in theory it would enable the use case the unit test is eluding to.